examples/fishjam-chat/screens/ConnectWithVideoRoomScreen.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Dimensions,
+  Image,
+  KeyboardAvoidingView,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { Button, TextInput, DismissKeyboard } from '../components';
+import {
+  AppRootStackParamList,
+  TabParamList,
+} from '../navigators/AppNavigator';
+import { joinRoomWithRoomManager } from '../utils/roomManager';
+
+type Props = CompositeScreenProps<
+  BottomTabScreenProps<TabParamList, 'ConnectWithVideoRoom'>,
+  NativeStackScreenProps<AppRootStackParamList>
+>;
+
+type VideoRoomEnv = 'staging' | 'prod';
+
+type VideoRoomData = {
+  env: VideoRoomEnv;
+  roomName: string;
+  userName: string;
+};
+
+async function saveStorageData(videoRoomData: VideoRoomData) {
+  await AsyncStorage.setItem('videoRoomData', JSON.stringify(videoRoomData));
+}
+
+async function readStorageData(): Promise<VideoRoomData> {
+  const storageData = await AsyncStorage.getItem('videoRoomData');
+  if (storageData) {
+    const videoRoomData = JSON.parse(storageData) as VideoRoomData;
+    return videoRoomData;
+  }
+  return { env: 'staging', roomName: '', userName: '' };
+}
+
+export function shouldShowVideoRoomTab() {
+  return (
+    !!process.env.EXPO_PUBLIC_VIDEOROOM_STAGING_ROOM_MANAGER &&
+    !!process.env.EXPO_PUBLIC_VIDEOROOM_PRODUCTION_ROOM_MANAGER
+  );
+}
+
+/**
+ * Connect with the VideoRoom - our example service for video conferences
+ */
+export default function ConnectScreen({ navigation }: Props) {
+  const [connectionError, setConnectionError] = useState<string | null>(null);
+  const [env, setEnv] = useState<VideoRoomEnv>('staging');
+  const [loading, setLoading] = useState(false);
+
+  const [roomName, setRoomName] = useState('');
+  const [userName, setUserName] = useState('');
+
+  useEffect(() => {
+    async function readData() {
+      const {
+        env: storedEnv,
+        roomName: storedRoomName,
+        userName: storedUserName,
+      } = await readStorageData();
+
+      setRoomName(storedRoomName);
+      setUserName(storedUserName);
+      setEnv(storedEnv);
+    }
+    readData();
+  }, []);
+
+  const onTapConnectButton = async () => {
+    try {
+      setConnectionError(null);
+      setLoading(true);
+      const roomManagerUrl =
+        env === 'staging'
+          ? process.env.EXPO_PUBLIC_VIDEOROOM_STAGING_ROOM_MANAGER!
+          : process.env.EXPO_PUBLIC_VIDEOROOM_PRODUCTION_ROOM_MANAGER!;
+      saveStorageData({ env, roomName, userName });
+
+      const { fishjamUrl, token } = await joinRoomWithRoomManager(
+        roomManagerUrl,
+        roomName,
+        userName,
+      );
+
+      navigation.navigate('Preview', {
+        userName,
+        fishjamUrl,
+        peerToken: token,
+      });
+    } catch (e) {
+      const message =
+        'message' in (e as Error) ? (e as Error).message : 'Unknown error';
```

Comment:
**suggestion:** Why not just `e.message ?? 'Unknown error'` Those type casts don't provide any additional value in this case, IMO.

Reply:
`e` is `unknown` type, so you have to cast first to ready `message` property.

Reply:
I don't have a strong option about it, but in that case, you could declare e as `any`. We don't really gain anything here from this typecast. 

If you would like to stick with typecast, I suggest a shorter version which seems more concise to me:
`const message = e instanceof Error ? e.message : 'Unknown error';` 

Both versions have, however, the disadvantage of passing empty/undefined messages only if the property is specified.

Reply:
:50_50:
My personal experience is that having `catch(error: any)` by default (`useUnknownInCatchVariables off`) could create additional problems. So as might create more code, in long term it is helpful.
And this is example app, so let's show good practices where we can ;)